core/primitives/src/action/mod.rs

```diff
@@ -106,6 +106,37 @@ impl fmt::Debug for DeployContractAction {
     }
 }
 
+/// Deploy global contract action
+#[serde_as]
+#[derive(
+    BorshSerialize,
+    BorshDeserialize,
+    serde::Serialize,
+    serde::Deserialize,
+    PartialEq,
+    Eq,
+    Clone,
+    ProtocolSchema,
+)]
+pub struct DeployGlobalContractAction {
+    /// WebAssembly binary
+    #[serde_as(as = "Base64")]
+    pub code: Vec<u8>,
+
+    /// Enables global contract code to referenced by owner account id
+    /// instead of code hash
+    pub link_account: bool,
```

Comment:
Rather than a bool, consider a field of bitflags. Those are more extensible without affecting the protocol as much in case we want to implement additional behaviours for the action.